groups/nts/ntsu/ntsu_socketutil.t.cpp

```diff
@@ -616,6 +709,221 @@ void testStreamSocketTransmissionBlob(ntsa::Transport::Value transport,
+
+    ntsa::Error error;
+
+    char DATA[] = "123456789";
+
+    bdlbb::SimpleBlobBufferFactory blobBufferFactory(3, allocator);
+
+    ntsa::Handle domesticSocket;
+    error = ntsu::SocketUtil::create(&domesticSocket, transport);
+    NTSCFG_TEST_ASSERT(!error);
+
+    error = ntsu::SocketUtil::bind(
+        ntsa::Endpoint(ntsa::LocalName::generateUnique()),
+        false,
+        domesticSocket);
+    NTSCFG_TEST_ASSERT(!error);
+
+    ntsa::Endpoint domesticSourceEndpoint;
+    error = ntsu::SocketUtil::sourceEndpoint(&domesticSourceEndpoint, 
+                                             domesticSocket);
+    NTSCFG_TEST_ASSERT(!error);
+
+    for (bsl::size_t iteration = 0; iteration < 2; ++iteration) {
+        NTSCFG_TEST_LOG_DEBUG << "Testing iteration " << iteration 
+                              << NTSCFG_TEST_LOG_END;
+
+        bdlbb::Blob clientBlob(&blobBufferFactory, allocator);
+        bdlbb::BlobUtil::append(&clientBlob, DATA, sizeof DATA - 1);
+
+        bdlbb::Blob serverBlob(&blobBufferFactory, allocator);
+        serverBlob.setLength(sizeof DATA - 1);
+        serverBlob.setLength(0);
+
+        // Enqueue outgoing data to transmit by the client socket.
+
+        {
+            ntsa::SendContext context;
+            ntsa::SendOptions options;
+
+            if (iteration == 0) {
+                options.setForeignHandle(domesticSocket);
+            }
+
+            error = ntsu::SocketUtil::send(
+                &context, clientBlob, options, client);
+            NTSCFG_TEST_ASSERT(!error);
+
+            NTSCFG_TEST_ASSERT(context.bytesSendable() == 9);
+            NTSCFG_TEST_ASSERT(context.bytesSent() == 9);
+        }
+
+        // Dequeue incoming data received by the server socket.
+
+        {
+            ntsa::ReceiveContext context;
+            ntsa::ReceiveOptions options;
+
+            if (iteration == 1) {
+                options.showForeignHandles();
+            }
+
+            error =
+                ntsu::SocketUtil::receive(
+                    &context, &serverBlob, options, server);
+            NTSCFG_TEST_ASSERT(!error);
+
+            NTSCFG_TEST_ASSERT(context.bytesReceivable() == 9);
+            NTSCFG_TEST_ASSERT(context.bytesReceived() == 9);
+
+            NTSCFG_TEST_ASSERT(serverBlob.length() == 9);
+            NTSCFG_TEST_ASSERT(bdlbb::BlobUtil::compare(serverBlob, clientBlob) 
+                               == 0);
+
+            if (iteration == 0) {
+                NTSCFG_TEST_TRUE(context.foreignHandle().isNull());
+            }
+            else if (iteration == 1) {
+                NTSCFG_TEST_TRUE(context.foreignHandle().isNull());
+
+                // Operating system implementation note: a file descriptor
+                // passed using socket control messages that is not "picked up"
+                // immediately when it is available is not subsequently
+                // retrievable with a subsequent call to 'recvmsg'.
+#if 0
```

Comment:
It seems like you forgot to remove this code. And the same is applicable to another (datagram based) test

Reply:
Yes this is dead code. I've deleted this preprocessed-out block in these two places.